bundles/org.eclipse.search.core/search/org/eclipse/search/internal/core/text/TextSearchEngineRegistry.java

```diff
@@ -63,25 +66,38 @@ private void updateEngine(String preferredId) {
 			// creation failed, clear preference
 			setPreferredEngineID(""); // set to default //$NON-NLS-1$
 		}
+		// empty string: default engine
 		fPreferredEngineId= ""; //$NON-NLS-1$
 		fPreferredEngine= TextSearchEngine.createDefault();
 	}
 
 	private String getPreferredEngineID() {
-		IPreferenceStore prefs= SearchPlugin.getDefault().getPreferenceStore();
-		String preferedEngine= prefs.getString(SearchPreferencePage.TEXT_SEARCH_ENGINE);
+		String pluginId = getPreferencePluginId();
+		String preferedEngine = Platform.getPreferencesService().get(PREFERENCE_ENGINE_KEY, null,
+				new IEclipsePreferences[] { InstanceScope.INSTANCE.getNode(pluginId) });
 		return preferedEngine;
 	}
 
 	private void setPreferredEngineID(String id) {
-		IPreferenceStore prefs= SearchPlugin.getDefault().getPreferenceStore();
-		prefs.setValue(SearchPreferencePage.TEXT_SEARCH_ENGINE, id);
+		String pluginId = getPreferencePluginId();
+		IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode(pluginId);
+		preferences.put(PREFERENCE_ENGINE_KEY, id);
+		try {
+			// forces the application to save the preferences
+			preferences.flush();
+		} catch (BackingStoreException e) {
+			e.printStackTrace();
+		}
+	}
+
+	protected String getPreferencePluginId() {
+		return "org.eclipse.search";
```

Comment:
So this is the thing you want to make configurable, isn't it? What could be alternative values? How JDT-LS should change it?

Reply:
I've updated this part of the code to now have a setter. An extender could abuse internal API (which jdt.ls has assured us they are fine doing) by calling SearchCorePlugin.getDefault().getTextSearchEngineRegistry().setPreferenceNodeId(val)